src/scripts/rerun_price_calculations.ts

```diff
@@ -0,0 +1,87 @@
+/*
+The Licensed Work is (c) 2023 Sygma
+SPDX-License-Identifier: LGPL-3.0-only
+*/
+import { caching } from "cache-manager"
+import { logger } from "../utils/logger"
+import { SharedConfig, getSharedConfig } from "../indexer/config"
+import TransferRepository from "../indexer/repository/transfer"
+import CoinMarketCapService from "../indexer/services/coinmarketcap/coinmarketcap.service"
+import TransfersService from "../services/transfers.service"
+
+const coinMarketCapAPIKey = process.env.COINMARKETCAP_API_KEY || ""
+const coinMarketCapUrl = process.env.COINMARKETCAP_API_URL || ""
+const ttlInMins = Number(process.env.CACHE_TTL_IN_MINS) || 5
+
+function getTokenSymbol(sharedConfig: SharedConfig, fromDomainId: number, resourceId: string): string {
+  const currentDomain = sharedConfig.domains.find(domain => domain.id == fromDomainId)
+  if (!currentDomain) {
+    throw new Error("Domain doesn't exist")
+  }
+  const currentResource = currentDomain.resources.find(resource => resource.resourceId == resourceId)
+  if (!currentResource) {
+    throw new Error("Resource doesn't exist")
+  }
+  return currentResource.symbol
+}
+
+async function rerunPriceCalculations(): Promise<void> {
+  const transfersService = new TransfersService()
+
+  const memoryCache = await caching("memory", {
+    ttl: ttlInMins * 1000,
+  })
+
+  const coinMarketCapServiceInstance = new CoinMarketCapService(coinMarketCapAPIKey, coinMarketCapUrl, memoryCache)
+  const transferRepository = new TransferRepository()
+
+  let transfers = []
+  const limit = 50
+  let page = 1
+  const sharedConfig = await getSharedConfig(process.env.SHARED_CONFIG_URL!)
+  for (;;) {
```

Comment:
nit: first time seeing this, would it be more clear to have while (condition)?

Reply:
The `for (;;)` command runs an infinite loop. There are few alternatives to it. One is to write a do-while loop with a specific condition but it would be needed to add at least one more condition check making it less clear, at least to me. Also, it is possible to write `while (true)` without any other code modifications but the problem is that  ESLint throws an error, I guess it is needed to set `checkLoops: true` in the config file (https://eslint.org/docs/latest/rules/no-constant-condition) but still making it unclear why does this type of infinite loop work (https://github.com/eslint/eslint/issues/5477)? Lastly, I can write a recursive function that does the same job but not sure if it would be more clear?

Reply:
does eslint throws error if using `while(flag)` instead of the plain `while(true)`?

Reply:
Oh yeah, it doesn't throw an error now. Then, the workaround would be to just have: `const flag = true; while(flag){}`. So instead of `for(;;)` I'll do it that way?

Reply:
I believe this is not as important. I would opt out for `while(true)`, but to make eslint happy we can go with `for(;;)`. I don't see a point in creating a variable `const flag = true;` if this value is never used/changed.

Reply:
I would resolve this then @wainola @mj52951 if nobody has something against that.